Tests/iaas/key-manager/check-for-key-manager.py

```diff
@@ -26,6 +27,53 @@ def connect(cloud_name: str) -> openstack.connection.Connection:
     )
 
 
+def check_for_member_role(conn: openstack.connection.Connection
+                          ) -> None:
+    """Method to check whether the current user has the member role.
+    :param connection:
+        The current connection to an OpenStack cloud.
+    :returns: boolean, when role with most priviledges is member
+    """
+
+    auth_data = conn.auth
+    auth_dict = {
+        "identity": {
+            "methods": ["password"],
+            "password": {
+                "user": {
+                    "name": auth_data['username'],
+                    "domain": {"name": auth_data['project_domain_name']},
+                    "password": auth_data['password']
+                }
+            },
+        },
+        "scope": {
+            "project": {
+                "domain": {"name": auth_data['project_domain_name']},
+                "name": auth_data['project_name']
+            }
+        }
+    }
+
+    has_member_role = False
+    request = conn.session.request(auth_data['auth_url'] + '/v3/auth/tokens',
+                                   'POST',
+                                   json={'auth': auth_dict})
+    for role in json.loads(request.content)["token"]["roles"]:
+        role_name = role["name"]
+        if role_name == "admin" or role_name == "manager":
```

Comment:
It is intended that the user has at maximum the member role. (I will change the docstring).

Yes if a CSP changes roles for users (actively) and all the corresponding policies to allow more than the member role does, and a test user has the member role and the custom role, we will get a false positive (check will pass) on this test. But the only other way (application credentials) failed within my deployment with the new secure RBAC. Unfortunately there is  currently no way for a simple user to ask OpenStack, which roles they have in a certain project.

Within the new secure RBAC we will only have the roles: reader, member, manager and admin available for users. I do not think, that these will change from OpenStack side soon, because the the work for secure RBAC is going on now for years and after adding the manager role, there was no change to role names.